lib/site_setting_extension.rb

```diff
@@ -952,10 +962,12 @@ def setting(name_arg, default = nil, opts = {})
 
       plugins[name] = opts[:plugin] if opts[:plugin]
 
-      type_supervisor.load_setting(
-        name,
-        opts.extract!(*SiteSettings::TypeSupervisor::CONSUMED_OPTS),
-      )
+      choices_opts = opts.extract!(*SiteSettings::TypeSupervisor::CONSUMED_OPTS)
+
+      i18n_dependent_choices[name] = choices_opts if opts[:i18n_dependent_choices]
+      if !opts[:i18n_dependent_choices] # || i18n keys are loaded
+        type_supervisor.load_setting(name, choices_opts)
+      end
```

Comment:
When all settings are getting loaded, if I18n is available, we want to just load_setting even if it's i18n dependent. The `||` statement here is missing as I am not sure how we can confirm the `server.xx.yml` file is loaded.

Doing `I18n.available_locales.includes?(:en)` is not accurate. `I18n.backend.instance_variable_get(:@translations)[:en].keys` evaluates to `[:number, :time, :faker]` before the file is loaded.